x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -693,6 +714,14 @@ LogEntryBuilder withOpaqueId(ThreadContext threadContext) {
             return this;
         }
 
+        LogEntryBuilder withForwardedFor(ThreadContext threadContext) {
+            final String forwardedFor = threadContext.getHeader(AuditTrail.X_FORWARDED_FOR_HEADER);
+            if (forwardedFor != null) {
+                with(FORWARDED_FOR_FIELD_NAME, forwardedFor.split("\\s+|,\\s*"));
```

Comment:
why allow splitting based on a space alone? the format should always be delimited by a comma followed by a space.

Reply:
Since ordering is important here, I guess we should leave it as a ugly string. It can be indexed so that it is searchable and in the raw format to preserve ordering info.

Also there is now a standard `Forwarded` header. Should we add support for that as well? It can be done separately

https://tools.ietf.org/html/rfc7239